pkg/controller/history/controller_history.go

```diff
@@ -351,6 +355,29 @@ func (rh *realHistory) ReleaseControllerRevision(parent metav1.Object, revision
 	return released, err
 }
 
+func generatePatchBytesForDelete(parentUID types.UID, revisionUID types.UID) ([]byte, error) {
```

Comment:
just a consideration: would be nice to create a general function out of these 2 and not have a similar logic in 3 places

https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/controller_ref_manager.go#L529-L547

https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/patch.go#L41-L56

Reply:
I have a question here: These two functions belong to different packages. Do you have suggestions which package should I build the common function into?

Thanks!

Reply:
controller/controller_utils.go seems like a good place, unless somebody has a better idea?

Reply:
I will get this done either this week or early next week; apologies.